### DISCUSSION TITLE
Can .chezmoiroot be executed as a template?

### DISCUSSION THREAD

**Discussion by Insecure88** [twpayne/chezmoi#2924] [Q&A]
I would like to have chezmoi point to two different root directories depending on the operating system.

**Comment by Lockszmith-GH** [comment]
Interesting idea, I doubt that is the case, as the templating engine might rely on data inside the chezmoi config file.

As it is mostly important during initialization - I am thinking there is a way 'around' this:
```
repo-root/
   os1-root/
   os2-root/
   .gitignore
   run_once_setup_.chezmoiroot.sh
   run_once_setup_.chezmoiroot.ps1
```

`.gitignore` will list `.chezmoiroot` so it will never be part of the repo, but the `run_once...` scripts will setup the environment correctly and then call `chezmoi init` which will reinitilize based on the new `.chezmoiroot`

This is just me thinking on my feet, never tested this though

**Reply by twpayne** [reply]
Indeed. `.chezmoiroot` is read before everything else, before even the source state is read, so template data is not available.
